src/lib.rs

```diff
@@ -360,6 +369,28 @@ pub struct PostTextChange<BlockNumber, Moment> {
     text: Vec<u8>,
 }
 
+bitflags! {
+    /// Represents a post reaction
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    #[derive(Encode, Decode, Default)]
+    pub struct PostReaction: u32 {
+        /// Represent nothing set
+        const NONE = 0b00000000;
+
+        /// Represent thumb up set
+        const UP = 0b00000001;
+
+        /// Represent thumb down set
+        const DOWN = 0b00000010;
+
+        /// Represent like set
+        const LIKE = 0b00000100;
+    }
+}
+
+/// Represents a poll identifier, each bit represent an item. multiple items select possible.
+pub type PollData = u64;
```

Comment:
The semantics of this type are too opaque, and name should be more descriptive.

Can you try doing this in an explicit type safe way, rather than encoding it in an integer.

Reply:
Ok, I will think about how to define the data structure like your comments.